tests/ComplexMappingsTests.cpp

```diff
@@ -280,6 +282,37 @@ TEST_F(ComplexMappingsTests, UInt32MappingsDeadband)
     }
 }
 
+TEST_F(ComplexMappingsTests, FrequencyFilter)
+{
+    const auto& outputType = _outputType::UINT32;
+    const auto& uintCombos = winningCombos[outputType];
+    for (const auto& combo : uintCombos)
+    {
+        const auto operationType = std::get<2>(combo);
+        const auto endian = endianForOperation.at(operationType);
+        const auto value = static_cast<uint32_t>(rand());
+        const auto bytes = wolkabout::DataParsers::uint32ToRegisters(value, endian);
+        std::chrono::milliseconds frequencyFilterValue = std::chrono::milliseconds(100);
+
+        const auto registerType = std::get<0>(combo);
+        auto mapping =
+          std::make_shared<wolkabout::UInt32Mapping>("TEST", registerType, std::vector<std::int32_t>{0, 1}, operationType, false, -1, 0.0, frequencyFilterValue);
+
+        EXPECT_FALSE(mapping->isInitialized());
+        EXPECT_FALSE(mapping->isValid());
+
+        EXPECT_NO_THROW(mapping->update(bytes));
+        EXPECT_EQ(value, mapping->getUint32Value());
+
+        EXPECT_TRUE(mapping->isInitialized());
+        EXPECT_TRUE(mapping->isValid());
+
+        EXPECT_FALSE(mapping->doesUpdate(wolkabout::DataParsers::uint32ToRegisters(value + 1, endian)));
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
Instead of sleeping, try setting the saved value to -100 to avoid delay

Reply:
I'd need to write a setter that will only be used in tests.
Currently, this test increases the total test duration from 8.8s to 9.6s
Do I still write it?

Reply:
You'd could actually override the access modifiers of class by writting "#define private public" before including Register mapping header, and thus being able to modify private members of class without the need for setters. We do it on other projects as well. But you may skip this for now as you only add delay on these two test cases, but it should be done if more delays are to be added

Reply:
Noted.
I'll leave it as it is right now, but will update if we add more tests.